addons/stock_sms/models/stock_picking.py

```diff
@@ -19,6 +19,8 @@ def _pre_action_done_hook(self):
 
     def _check_warn_sms(self):
         warn_sms_pickings = self.browse()
+        if self.env.context.get('skip_sms'):
+            return warn_sms_pickings
```

Comment:
if you add a skip_* context key, could you add at a consistent place?
if you look at stock picking _pre_action_done_hook
``` python
    def _pre_action_done_hook(self):
        if not self.env.context.get('skip_immediate'):
            pickings_to_immediate = self._check_immediate()
            if pickings_to_immediate:
                return pickings_to_immediate._action_generate_immediate_wizard(show_transfers=self._should_show_transfers())

        if not self.env.context.get('skip_backorder'):
            pickings_to_backorder = self._check_backorder()
            if pickings_to_backorder:
                return pickings_to_backorder._action_generate_backorder_wizard(show_transfers=self._should_show_transfers())
        return True
```
so it seems to me the right place would be in the same method which currently looks like it

``` python
    def _pre_action_done_hook(self):
        res = super()._pre_action_done_hook()
        if res is True:
            pickings_to_warn_sms = self._check_warn_sms()
            if pickings_to_warn_sms:
                return pickings_to_warn_sms._action_generate_warn_sms_wizard()
        return res
```